app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+          }
+
+          override fun onZoom(event: ZoomEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+        })
+  }
+
+  Box(modifier = Modifier.fillMaxSize()) {
+    AndroidView(
+        factory = { mapView }, modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_MAP))
+
+    IconButton(
+        onClick = {
+          // TODO : Adapt the map screen to navigation
+          Toast.makeText(context, "Menu not implemented yet", Toast.LENGTH_SHORT).show()
+        },
+        modifier =
+            Modifier.padding(16.dp)
+                .align(Alignment.TopStart)
+                // Clip needs to be before background
+                .clip(RoundedCornerShape(8.dp))
+                .background(MaterialTheme.colorScheme.surface)
+                .testTag(MapScreen.TEST_TAG_MENU_BUTTON)) {
+          Icon(
+              Icons.Default.Menu,
+              contentDescription =
+                  context.getString(R.string.map_screen_menu_button_content_description),
+              tint = MaterialTheme.colorScheme.onSurface)
+        }
+
+    CollapsibleHikesList(hikingRoutesViewModel)
+  }
+
+  // Initialize the list of hiking routes once when the map is loaded
+  // Leave subsequent updates to the map listener
+  LaunchedEffect(Unit) { hikingRoutesViewModel.setArea(mapView.boundingBox) }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CollapsibleHikesList(hikingRoutesViewModel: ListOfHikeRoutesViewModel) {
+  val scaffoldState = rememberBottomSheetScaffoldState()
+  val routes = hikingRoutesViewModel.hikeRoutes.collectAsState()
+  val context = LocalContext.current
+
+  BottomSheetScaffold(
+      scaffoldState = scaffoldState,
+      sheetContainerColor = MaterialTheme.colorScheme.surface,
+      sheetContent = {
+        Column(modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_HIKES_LIST)) {
+          LazyColumn(modifier = Modifier.fillMaxSize()) {
+            if (routes.value.isEmpty()) {
+              item {
+                // Use a box to center the Text composable of the empty list message
+                Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+                  Text(
+                      text = context.getString(R.string.map_screen_empty_hikes_list_message),
+                      style = MaterialTheme.typography.bodyLarge,
+                      // Align the text within the Text composable to the center
+                      textAlign = TextAlign.Center)
+                }
+              }
+            } else {
+              items(routes.value.size) { index: Int ->
+                HikingRouteItem(
+                    title = routes.value[index],
+                    altitudeDifference = 1000,
+                    isSuitable = index % 2 == 0,
+                    onClick = {
+                      // TODO : Navigate to the details of the selected hiking route
```

Comment:
Warning: Another todo here, make sure to not leave this in the code and rather add tasks to the backlog

Reply:
Fair, however this task would be a lot of elements :
- Create the Hike details screen in Figma
- Implement the Hike details screen -> Implement navigation from map screen to hike details screen

So I'm not sure I should create all of those tasks already, maybe keeping the TODO is better.

Reply:
I'm afraid it might get lost in the code, and a task would ensure oversight, and ensure that all our upcoming tasks are in one place. Perhaps its enough to summarise all these tasks in one task for now, and the SCRUM master can figure out how to best split them up?

Reply:
Indeed, putting all of the tasks in one place is the best option. I'll create the tasks now.

Reply:
Created [this draft task](https://github.com/orgs/HikeMate/projects/1/views/2?pane=issue&itemId=83430191), what do you think @davidholzwarth?